### ISSUE TITLE
Sumo compile error: libssl.so: file not recognized: File format not recognized

### ISSUE THREAD

**Issue by geoffrey-vl** [Azure/meta-iotedge#34]
**Describe the bug**
I'm trying to get the Sumo branch to build on Yocto Rocko, since it is closest to Rocko and best maintained. During compiling it fails to find `libssl` for what I understand: `libssl.so: file not recognized: File format not recognized`.

Its probable related to https://github.com/Azure/meta-iotedge/issues/22#issuecomment-546474794

**To Reproduce**
1. add meta-vertualization
2. add meta-rust (version 9487b089ea4779c2b494b17b9254219226efa539)
3. add meta-iotedge
4. bitbake iotedge-cli

**Expected behavior**
Should compile and package

**Logs**
```
| #Start building HSM dev-mode library
| detected home dir change, cleaning out entire build directory
| running: "cmake" "/home/jenkins/build/build-x11-pico-imx7/tmp/work/cortexa7hf-neon-poky-linux-gnueabi/iotedge-cli/1.0.8-r0/iotedge-1.0.8/edgelet/hsm-sys/azure-iot-hsm-c" "-Duse_openssl=ON" "-DCMAKE_BUILD_TYPE=Release" "-Drun_unittests=ON" "-Duse_default_uuid=ON" "-Duse_http=OFF" "-Dskip_samples=ON" "-Drun_valgrind=OFF" "-Duse_emulator=OFF" "-DBUILD_SHARED=ON" "-DUSE_TEST_TPM_INTERFACE_IN_MEM=OFF" "-DCMAKE_INSTALL_PREFIX=/home/jenkins/build/build-x11-pico-imx7/tmp/work/cortexa7hf-neon-poky-linux-gnueabi/iotedge-cli/1.0.8-r0/iotedge-1.0.8/edgelet/target/arm-poky-linux-gnueabi/release/build/hsm-sys-fe99d18cce916c08/out" "-DCMAKE_C_FLAGS= -ffunction-sections -fdata-sections -fPIC -pipe -feliminate-unused-debug-types -fdebug-prefix-map=/home/jenkins/build/build-x11-pico-imx7/tmp/work/cortexa7hf-neon-poky-linux-gnueabi/iotedge-cli/1.0.8-r0=/usr/src/debug/iotedge-cli/1.0.8-r0 -fdebug-prefix-map=/home/jenkins/build/build-x11-pico-imx7/tmp/work/cortexa7hf-neon-poky-linux-gnueabi/iotedge-cli/1.0.8-r0/recipe-sysroot-native= -fdebug-prefix-map=/home/jenkins/build/build-x11-pico-imx7/tmp/work/cortexa7hf-neon-poky-linux-gnueabi/iotedge-cli/1.0.8-r0/recipe-sysroot=" "-DCMAKE_C_COMPILER=/home/jenkins/build/build-x11-pico-imx7/tmp/work/cortexa7hf-neon-poky-linux-gnueabi/iotedge-cli/1.0.8-r0/wrapper/target-rust-cc" "-DCMAKE_CXX_FLAGS= -ffunction-sections -fdata-sections -fPIC -pipe -feliminate-unused-debug-types -fdebug-prefix-map=/home/jenkins/build/build-x11-pico-imx7/tmp/work/cortexa7hf-neon-poky-linux-gnueabi/iotedge-cli/1.0.8-r0=/usr/src/debug/iotedge-cli/1.0.8-r0 -fdebug-prefix-map=/home/jenkins/build/build-x11-pico-imx7/tmp/work/cortexa7hf-neon-poky-linux-gnueabi/iotedge-cli/1.0.8-r0/recipe-sysroot-native= -fdebug-prefix-map=/home/jenkins/build/build-x11-pico-imx7/tmp/work/cortexa7hf-neon-poky-linux-gnueabi/iotedge-cli/1.0.8-r0/recipe-sysroot= -fvisibility-inlines-hidden" "-DCMAKE_CXX_COMPILER=/home/jenkins/build/build-x11-pico-imx7/tmp/work/cortexa7hf-neon-poky-linux-gnueabi/iotedge-cli/1.0.8-r0/wrapper/target-rust-cxx"
| -- The C compiler identification is GNU 7.3.0
| -- The CXX compiler identification is GNU 7.3.0
| -- Check for working C compiler: /home/jenkins/build/build-x11-pico-imx7/tmp/work/cortexa7hf-neon-poky-linux-gnueabi/iotedge-cli/1.0.8-r0/wrapper/target-rust-cc
| -- Check for working C compiler: /home/jenkins/build/build-x11-pico-imx7/tmp/work/cortexa7hf-neon-poky-linux-gnueabi/iotedge-cli/1.0.8-r0/wrapper/target-rust-cc -- works
| -- Detecting C compiler ABI info
| -- Detecting C compiler ABI info - done
| -- Detecting C compile features
| -- Detecting C compile features - done
| -- Check for working CXX compiler: /home/jenkins/build/build-x11-pico-imx7/tmp/work/cortexa7hf-neon-poky-linux-gnueabi/iotedge-cli/1.0.8-r0/wrapper/target-rust-cxx
| -- Check for working CXX compiler: /home/jenkins/build/build-x11-pico-imx7/tmp/work/cortexa7hf-neon-poky-linux-gnueabi/iotedge-cli/1.0.8-r0/wrapper/target-rust-cxx -- works
| -- Detecting CXX compiler ABI info
| -- Detecting CXX compiler ABI info - done
| -- Detecting CXX compile features
| -- Detecting CXX compile features - done
| -- Found OpenSSL: /home/jenkins/build/build-x11-pico-imx7/tmp/work/cortexa7hf-neon-poky-linux-gnueabi/iotedge-cli/1.0.8-r0/recipe-sysroot-native/usr/lib/libcrypto.so (found version "1.0.2n")
| -- Looking for include file stdint.h
| -- Looking for include file stdint.h - found
| -- Looking for include file stdbool.h
| -- Looking for include file stdbool.h - found
| -- target architecture: ARM
| -- Performing Test CXX_FLAG_CXX11
| -- Performing Test CXX_FLAG_CXX11 - Success
| -- target architecture: ARM
| -- Configuring done
| -- Generating done
| -- Build files have been written to: /home/jenkins/build/build-x11-pico-imx7/tmp/work/cortexa7hf-neon-poky-linux-gnueabi/iotedge-cli/1.0.8-r0/iotedge-1.0.8/edgelet/target/arm-poky-linux-gnueabi/release/build/hsm-sys-fe99d18cce916c08/out/build
| running: "cmake" "--build" "." "--target" "install" "--config" "Release" "--"
| Scanning dependencies of target aziotsharedutil
| [  0%] Building C object deps/c-shared/CMakeFiles/aziotsharedutil.dir/src/azure_base32.c.o
| [  0%] Building C object deps/c-shared/CMakeFiles/aziotsharedutil.dir/src/azure_base64.c.o
| [  1%] Building C object deps/c-shared/CMakeFiles/aziotsharedutil.dir/src/buffer.c.o
| [  1%] Building C object deps/c-shared/CMakeFiles/aziotsharedutil.dir/src/constbuffer_array.c.o
| [  2%] Building C object deps/c-shared/CMakeFiles/aziotsharedutil.dir/src/connection_string_parser.c.o
| [  2%] Building C object deps/c-shared/CMakeFiles/aziotsharedutil.dir/src/constbuffer.c.o
| [  3%] Building C object deps/c-shared/CMakeFiles/aziotsharedutil.dir/src/consolelogger.c.o
| Scanning dependencies of target ctest
| Scanning dependencies of target testrunnerswitcher
| [  3%] Building C object tests/c-shared/testtools/ctest/CMakeFiles/ctest.dir/src/ctest.c.o
| [  4%] Building C object tests/c-shared/testtools/testrunner/CMakeFiles/testrunnerswitcher.dir/src/testmutex.c.o
| [  4%] Building C object tests/c-shared/testtools/testrunner/CMakeFiles/testrunnerswitcher.dir/src/ctrs_sprintf.c.o
| [  4%] Building C object deps/c-shared/CMakeFiles/aziotsharedutil.dir/src/crt_abstractions.c.o
| [  5%] Linking C static library libtestrunnerswitcher.a
| [  5%] Building C object deps/c-shared/CMakeFiles/aziotsharedutil.dir/src/constmap.c.o
| [  5%] Built target testrunnerswitcher
| [  6%] Building C object deps/c-shared/CMakeFiles/aziotsharedutil.dir/src/doublylinkedlist.c.o
| Scanning dependencies of target umock_c
| [  6%] Building C object deps/c-shared/deps/umock-c/CMakeFiles/umock_c.dir/src/umock_c.c.o
| [  6%] Building C object deps/c-shared/CMakeFiles/aziotsharedutil.dir/src/gballoc.c.o
| [  7%] Linking C static library libctest.a
| [  7%] Built target ctest
| [  8%] Building C object deps/c-shared/CMakeFiles/aziotsharedutil.dir/src/gbnetwork.c.o
| [  8%] Building C object deps/c-shared/deps/umock-c/CMakeFiles/umock_c.dir/src/umock_c_negative_tests.c.o
| [  9%] Building C object deps/c-shared/deps/umock-c/CMakeFiles/umock_c.dir/src/umockalloc.c.o
| [  9%] Building C object deps/c-shared/CMakeFiles/aziotsharedutil.dir/src/gb_stdio.c.o
| [  9%] Building C object deps/c-shared/deps/umock-c/CMakeFiles/umock_c.dir/src/umockcall.c.o
| [ 10%] Building C object deps/c-shared/CMakeFiles/aziotsharedutil.dir/src/gb_time.c.o
| [ 10%] Building C object deps/c-shared/CMakeFiles/aziotsharedutil.dir/src/gb_rand.c.o
| [ 10%] Building C object deps/c-shared/CMakeFiles/aziotsharedutil.dir/src/hmac.c.o
| [ 11%] Building C object deps/c-shared/CMakeFiles/aziotsharedutil.dir/src/hmacsha256.c.o
| [ 12%] Building C object deps/c-shared/deps/umock-c/CMakeFiles/umock_c.dir/src/umockcallrecorder.c.o
| [ 12%] Building C object deps/c-shared/CMakeFiles/aziotsharedutil.dir/src/xio.c.o
| [ 13%] Building C object deps/c-shared/CMakeFiles/aziotsharedutil.dir/src/singlylinkedlist.c.o
| [ 13%] Building C object deps/c-shared/deps/umock-c/CMakeFiles/umock_c.dir/src/umocktypename.c.o
| [ 13%] Building C object deps/c-shared/CMakeFiles/aziotsharedutil.dir/src/map.c.o
| [ 14%] Building C object deps/c-shared/deps/umock-c/CMakeFiles/umock_c.dir/src/umocktypes.c.o
| [ 14%] Building C object deps/c-shared/deps/umock-c/CMakeFiles/umock_c.dir/src/umocktypes_bool.c.o
| [ 14%] Building C object deps/c-shared/deps/umock-c/CMakeFiles/umock_c.dir/src/umocktypes_c.c.o
| [ 15%] Building C object deps/c-shared/deps/umock-c/CMakeFiles/umock_c.dir/src/umocktypes_stdint.c.o
| [ 16%] Building C object deps/c-shared/CMakeFiles/aziotsharedutil.dir/src/sastoken.c.o
| [ 16%] Building C object deps/c-shared/deps/umock-c/CMakeFiles/umock_c.dir/src/umocktypes_charptr.c.o
| [ 17%] Building C object deps/c-shared/deps/umock-c/CMakeFiles/umock_c.dir/src/umockcallpairs.c.o
| [ 17%] Building C object deps/c-shared/CMakeFiles/aziotsharedutil.dir/src/sha1.c.o
| [ 17%] Building C object deps/c-shared/deps/umock-c/CMakeFiles/umock_c.dir/src/umockstring.c.o
| [ 18%] Building C object deps/c-shared/deps/umock-c/CMakeFiles/umock_c.dir/src/umockautoignoreargs.c.o
| [ 18%] Building C object deps/c-shared/deps/umock-c/CMakeFiles/umock_c.dir/src/umock_log.c.o
| [ 18%] Building C object deps/c-shared/CMakeFiles/aziotsharedutil.dir/src/sha224.c.o
| [ 19%] Building C object deps/c-shared/CMakeFiles/aziotsharedutil.dir/src/sha384-512.c.o
| [ 19%] Linking C static library libumock_c.a
| [ 19%] Built target umock_c
| [ 19%] Building C object deps/c-shared/CMakeFiles/aziotsharedutil.dir/src/strings.c.o
| [ 20%] Building C object deps/c-shared/CMakeFiles/aziotsharedutil.dir/src/string_token.c.o
| Scanning dependencies of target hsm_certificate_props_ut_exe
| Scanning dependencies of target hsm_tpm_select_ut_exe
| [ 20%] Building C object tests/hsm_certificate_props_ut/CMakeFiles/hsm_certificate_props_ut_exe.dir/hsm_certificate_props_ut.c.o
| [ 20%] Building C object tests/hsm_tpm_select_ut/CMakeFiles/hsm_tpm_select_ut_exe.dir/hsm_tpm_select_ut.c.o
| [ 21%] Building C object tests/hsm_certificate_props_ut/CMakeFiles/hsm_certificate_props_ut_exe.dir/__/__/src/hsm_certificate_props.c.o
| [ 21%] Building C object deps/c-shared/CMakeFiles/aziotsharedutil.dir/src/string_tokenizer.c.o
| [ 22%] Building C object deps/c-shared/CMakeFiles/aziotsharedutil.dir/src/uuid.c.o
| [ 22%] Building C object deps/c-shared/CMakeFiles/aziotsharedutil.dir/src/urlencode.c.o
| [ 22%] Building C object tests/hsm_certificate_props_ut/CMakeFiles/hsm_certificate_props_ut_exe.dir/__/__/deps/c-shared/src/xlogging.c.o
| [ 23%] Building C object tests/hsm_certificate_props_ut/CMakeFiles/hsm_certificate_props_ut_exe.dir/__/__/deps/c-shared/src/consolelogger.c.o
| [ 23%] Building C object deps/c-shared/CMakeFiles/aziotsharedutil.dir/src/usha.c.o
| [ 23%] Building C object tests/hsm_certificate_props_ut/CMakeFiles/hsm_certificate_props_ut_exe.dir/__/__/deps/c-shared/tests/real_test_files/real_crt_abstractions.c.o
| [ 24%] Building C object deps/c-shared/CMakeFiles/aziotsharedutil.dir/src/vector.c.o
| [ 24%] Building C object deps/c-shared/CMakeFiles/aziotsharedutil.dir/src/xlogging.c.o
| [ 25%] Building C object deps/c-shared/CMakeFiles/aziotsharedutil.dir/src/optionhandler.c.o
| [ 25%] Building C object deps/c-shared/CMakeFiles/aziotsharedutil.dir/src/memory_data.c.o
| Scanning dependencies of target certificate_info_ut_exe
| [ 25%] Building C object tests/certificate_info_ut/CMakeFiles/certificate_info_ut_exe.dir/certificate_info_ut.c.o
| Scanning dependencies of target edge_hsm_tpm_ut_exe
| [ 25%] Building C object tests/edge_hsm_tpm_ut/CMakeFiles/edge_hsm_tpm_ut_exe.dir/__/__/src/hsm_client_tpm_in_mem.c.o
| [ 25%] Building C object tests/certificate_info_ut/CMakeFiles/certificate_info_ut_exe.dir/pki_mocked.c.o
| [ 26%] Building C object deps/c-shared/CMakeFiles/aziotsharedutil.dir/adapters/agenttime.c.o
| [ 26%] Building C object deps/c-shared/CMakeFiles/aziotsharedutil.dir/adapters/condition_pthreads.c.o
| [ 27%] Building C object tests/edge_hsm_tpm_ut/CMakeFiles/edge_hsm_tpm_ut_exe.dir/__/__/src/hsm_log.c.o
| [ 27%] Building C object tests/edge_hsm_tpm_ut/CMakeFiles/edge_hsm_tpm_ut_exe.dir/__/__/src/constants.c.o
| [ 28%] Building C object tests/edge_hsm_tpm_ut/CMakeFiles/edge_hsm_tpm_ut_exe.dir/edge_hsm_tpm_ut.c.o
| [ 28%] Building C object deps/c-shared/CMakeFiles/aziotsharedutil.dir/adapters/lock_pthreads.c.o
| [ 29%] Building C object deps/c-shared/CMakeFiles/aziotsharedutil.dir/adapters/platform_linux.c.o
| [ 29%] Building C object tests/hsm_tpm_select_ut/CMakeFiles/hsm_tpm_select_ut_exe.dir/__/__/src/hsm_client_tpm_select.c.o
| Scanning dependencies of target edge_hsm_key_intf_sas_ut_exe
| [ 30%] Building C object tests/edge_hsm_key_intf_sas_ut/CMakeFiles/edge_hsm_key_intf_sas_ut_exe.dir/__/__/src/edge_hsm_key_interface.c.o
| [ 31%] Building C object tests/hsm_certificate_props_ut/CMakeFiles/hsm_certificate_props_ut_exe.dir/main.c.o
| [ 32%] Building C object tests/certificate_info_ut/CMakeFiles/certificate_info_ut_exe.dir/__/__/src/hsm_log.c.o
| [ 32%] Building C object deps/c-shared/CMakeFiles/aziotsharedutil.dir/adapters/socketio_berkeley.c.o
| [ 32%] Building C object tests/edge_hsm_key_intf_sas_ut/CMakeFiles/edge_hsm_key_intf_sas_ut_exe.dir/__/__/src/edge_sas_perform_sign_with_key.c.o
| [ 32%] Building C object tests/edge_hsm_key_intf_sas_ut/CMakeFiles/edge_hsm_key_intf_sas_ut_exe.dir/__/__/src/edge_sas_key.c.o
| [ 33%] Building C object tests/edge_hsm_key_intf_sas_ut/CMakeFiles/edge_hsm_key_intf_sas_ut_exe.dir/__/__/src/hsm_log.c.o
| [ 33%] Building C object tests/edge_hsm_key_intf_sas_ut/CMakeFiles/edge_hsm_key_intf_sas_ut_exe.dir/__/__/src/constants.c.o
| [ 34%] Building C object tests/edge_hsm_key_intf_sas_ut/CMakeFiles/edge_hsm_key_intf_sas_ut_exe.dir/edge_hsm_key_intf_sas_ut.c.o
| [ 34%] Building C object tests/edge_hsm_key_intf_sas_ut/CMakeFiles/edge_hsm_key_intf_sas_ut_exe.dir/main.c.o
| [ 35%] Building C object tests/hsm_tpm_select_ut/CMakeFiles/hsm_tpm_select_ut_exe.dir/__/__/src/hsm_log.c.o
| [ 35%] Building C object tests/certificate_info_ut/CMakeFiles/certificate_info_ut_exe.dir/main.c.o
| [ 35%] Building C object tests/hsm_tpm_select_ut/CMakeFiles/hsm_tpm_select_ut_exe.dir/__/__/src/constants.c.o
| [ 36%] Building C object tests/hsm_tpm_select_ut/CMakeFiles/hsm_tpm_select_ut_exe.dir/main.c.o
| Scanning dependencies of target edge_hsm_crypto_ut_exe
| [ 36%] Building C object tests/edge_hsm_crypto_ut/CMakeFiles/edge_hsm_crypto_ut_exe.dir/__/__/src/edge_hsm_client_crypto.c.o
| Scanning dependencies of target edge_openssl_common_ut_exe
| [ 36%] Building C object tests/edge_openssl_common_ut/CMakeFiles/edge_openssl_common_ut_exe.dir/edge_openssl_common_ut.c.o
| [ 36%] Building C object tests/edge_hsm_crypto_ut/CMakeFiles/edge_hsm_crypto_ut_exe.dir/__/__/src/hsm_log.c.o
| [ 37%] Building C object tests/edge_hsm_crypto_ut/CMakeFiles/edge_hsm_crypto_ut_exe.dir/__/__/src/constants.c.o
| [ 37%] Building C object tests/edge_hsm_crypto_ut/CMakeFiles/edge_hsm_crypto_ut_exe.dir/edge_hsm_crypto_ut.c.o
| [ 38%] Building C object deps/c-shared/CMakeFiles/aziotsharedutil.dir/adapters/tickcounter_linux.c.o
| [ 39%] Building C object tests/edge_hsm_crypto_ut/CMakeFiles/edge_hsm_crypto_ut_exe.dir/main.c.o
| [ 39%] Building C object deps/c-shared/CMakeFiles/aziotsharedutil.dir/adapters/threadapi_pthreads.c.o
| [ 40%] Building C object deps/c-shared/CMakeFiles/aziotsharedutil.dir/adapters/uniqueid_stub.c.o
| [ 40%] Building C object deps/c-shared/CMakeFiles/aziotsharedutil.dir/adapters/envvariable.c.o
| [ 40%] Building C object deps/c-shared/CMakeFiles/aziotsharedutil.dir/adapters/linux_time.c.o
| [ 41%] Building C object deps/c-shared/CMakeFiles/aziotsharedutil.dir/src/http_proxy_io.c.o
| [ 41%] Building C object deps/c-shared/CMakeFiles/aziotsharedutil.dir/adapters/tlsio_openssl.c.o
| [ 42%] Building C object deps/c-shared/CMakeFiles/aziotsharedutil.dir/adapters/x509_openssl.c.o
| [ 42%] Building C object deps/c-shared/CMakeFiles/aziotsharedutil.dir/src/wsio.c.o
| [ 43%] Building C object deps/c-shared/CMakeFiles/aziotsharedutil.dir/src/uws_client.c.o
| [ 43%] Building C object deps/c-shared/CMakeFiles/aziotsharedutil.dir/src/uws_frame_encoder.c.o
| [ 43%] Building C object deps/c-shared/CMakeFiles/aziotsharedutil.dir/src/utf8_checker.c.o
| [ 44%] Building C object deps/c-shared/CMakeFiles/aziotsharedutil.dir/src/ws_url.c.o
| [ 44%] Building C object tests/edge_openssl_common_ut/CMakeFiles/edge_openssl_common_ut_exe.dir/openssl_mocked.c.o
| [ 45%] Building C object tests/edge_openssl_common_ut/CMakeFiles/edge_openssl_common_ut_exe.dir/main.c.o
| Scanning dependencies of target edge_openssl_pki_ut_exe
| [ 45%] Building C object tests/edge_openssl_pki_ut/CMakeFiles/edge_openssl_pki_ut_exe.dir/edge_openssl_pki_ut.c.o
| [ 45%] Linking C static library libaziotsharedutil.a
| Scanning dependencies of target hsm_client_tpm_ut_exe
| [ 45%] Building C object tests/hsm_client_tpm_ut/CMakeFiles/hsm_client_tpm_ut_exe.dir/hsm_client_tpm_ut.c.o
| [ 45%] Built target aziotsharedutil
| [ 45%] Building C object tests/edge_hsm_tpm_ut/CMakeFiles/edge_hsm_tpm_ut_exe.dir/main.c.o
| [ 46%] Building C object tests/edge_openssl_pki_ut/CMakeFiles/edge_openssl_pki_ut_exe.dir/pki_mocked.c.o
| [ 46%] Linking C executable hsm_tpm_select_ut_exe
| [ 46%] Built target hsm_tpm_select_ut_exe
| Scanning dependencies of target edge_openssl_enc_ut_exe
| [ 46%] Building C object tests/edge_openssl_enc_ut/CMakeFiles/edge_openssl_enc_ut_exe.dir/__/__/src/edge_enc_openssl_key.c.o
| [ 47%] Building C object tests/edge_openssl_enc_ut/CMakeFiles/edge_openssl_enc_ut_exe.dir/__/__/src/hsm_log.c.o
| [ 47%] Building C object tests/edge_openssl_enc_ut/CMakeFiles/edge_openssl_enc_ut_exe.dir/edge_openssl_enc_ut.c.o
| [ 47%] Building C object tests/edge_openssl_pki_ut/CMakeFiles/edge_openssl_pki_ut_exe.dir/__/__/src/hsm_log.c.o
| [ 47%] Building C object tests/edge_openssl_pki_ut/CMakeFiles/edge_openssl_pki_ut_exe.dir/main.c.o
| [ 47%] Building C object tests/edge_openssl_enc_ut/CMakeFiles/edge_openssl_enc_ut_exe.dir/main.c.o
| [ 48%] Building C object tests/hsm_client_tpm_ut/CMakeFiles/hsm_client_tpm_ut_exe.dir/__/__/src/hsm_client_tpm_device.c.o
| [ 48%] Building C object tests/hsm_client_tpm_ut/CMakeFiles/hsm_client_tpm_ut_exe.dir/__/__/src/hsm_log.c.o
| [ 49%] Building C object tests/hsm_client_tpm_ut/CMakeFiles/hsm_client_tpm_ut_exe.dir/__/__/src/constants.c.o
| [ 49%] Building C object tests/hsm_client_tpm_ut/CMakeFiles/hsm_client_tpm_ut_exe.dir/main.c.o
| [ 49%] Linking C executable edge_openssl_common_ut_exe
| Scanning dependencies of target edge_hsm_x509_ut_exe
| [ 49%] Building C object tests/edge_hsm_x509_ut/CMakeFiles/edge_hsm_x509_ut_exe.dir/__/__/src/edge_hsm_client_x509.c.o
| [ 49%] Built target edge_openssl_common_ut_exe
| Scanning dependencies of target utpm
| [ 50%] Building C object deps/utpm/CMakeFiles/utpm.dir/src/Marshal.c.o
| [ 51%] Building C object tests/edge_hsm_x509_ut/CMakeFiles/edge_hsm_x509_ut_exe.dir/__/__/src/hsm_log.c.o
| [ 51%] Building C object tests/edge_hsm_x509_ut/CMakeFiles/edge_hsm_x509_ut_exe.dir/__/__/src/constants.c.o
| [ 52%] Building C object tests/edge_hsm_x509_ut/CMakeFiles/edge_hsm_x509_ut_exe.dir/edge_hsm_x509_ut.c.o
| [ 52%] Linking C executable hsm_certificate_props_ut_exe
| [ 52%] Built target hsm_certificate_props_ut_exe
| [ 52%] Building C object deps/utpm/CMakeFiles/utpm.dir/src/Memory.c.o
| [ 53%] Building C object deps/utpm/CMakeFiles/utpm.dir/src/tpm_codec.c.o
| [ 53%] Building C object deps/utpm/CMakeFiles/utpm.dir/src/gbfiledescript.c.o
| [ 54%] Building C object deps/utpm/CMakeFiles/utpm.dir/src/tpm_comm_linux.c.o
| [ 54%] Building C object deps/utpm/CMakeFiles/utpm.dir/src/tpm_socket_comm.c.o
| [ 54%] Building C object tests/edge_hsm_x509_ut/CMakeFiles/edge_hsm_x509_ut_exe.dir/main.c.o
| [ 55%] Linking C static library libutpm.a
| [ 55%] Built target utpm
| Scanning dependencies of target iothsm
| [ 56%] Building C object CMakeFiles/iothsm.dir/src/certificate_info.c.o
| [ 56%] Building C object CMakeFiles/iothsm.dir/src/constants.c.o
| [ 56%] Building C object CMakeFiles/iothsm.dir/src/edge_enc_openssl_key.c.o
| [ 57%] Building C object CMakeFiles/iothsm.dir/src/edge_hsm_client_crypto.c.o
| [ 57%] Building C object CMakeFiles/iothsm.dir/src/edge_hsm_client_store.c.o
| [ 58%] Building C object CMakeFiles/iothsm.dir/src/edge_hsm_client_x509.c.o
| [ 58%] Building C object CMakeFiles/iothsm.dir/src/edge_hsm_key_interface.c.o
| [ 59%] Building C object CMakeFiles/iothsm.dir/src/edge_openssl_common.c.o
| [ 59%] Building C object CMakeFiles/iothsm.dir/src/edge_sas_perform_sign_with_key.c.o
| [ 59%] Building C object CMakeFiles/iothsm.dir/src/edge_pki_openssl.c.o
| [ 60%] Building C object CMakeFiles/iothsm.dir/src/edge_sas_key.c.o
| [ 60%] Building C object CMakeFiles/iothsm.dir/src/hsm_certificate_props.c.o
| [ 61%] Building C object CMakeFiles/iothsm.dir/src/hsm_client_data.c.o
| [ 61%] Building C object CMakeFiles/iothsm.dir/src/hsm_client_tpm_device.c.o
| [ 62%] Linking C executable certificate_info_ut_exe
| [ 63%] Building C object CMakeFiles/iothsm.dir/src/hsm_client_tpm_in_mem.c.o
| [ 63%] Built target certificate_info_ut_exe
| [ 63%] Building C object CMakeFiles/iothsm.dir/src/hsm_client_tpm_select.c.o
| [ 63%] Building C object CMakeFiles/iothsm.dir/src/hsm_log.c.o
| [ 64%] Building C object CMakeFiles/iothsm.dir/src/hsm_utils.c.o
| [ 64%] Linking C shared library libiothsm.so
| CMakeFiles/iothsm.dir/build.make:543: recipe for target 'libiothsm.so.1.0.8' failed
| CMakeFiles/Makefile2:68: recipe for target 'CMakeFiles/iothsm.dir/all' failed
| [ 65%] Linking C executable edge_hsm_key_intf_sas_ut_exe
| [ 65%] Built target edge_hsm_key_intf_sas_ut_exe
| [ 66%] Linking C executable edge_openssl_enc_ut_exe
| [ 66%] Built target edge_openssl_enc_ut_exe
| [ 66%] Linking C executable edge_hsm_tpm_ut_exe
| [ 66%] Built target edge_hsm_tpm_ut_exe
| [ 66%] Linking C executable edge_hsm_crypto_ut_exe
| [ 66%] Built target edge_hsm_crypto_ut_exe
| [ 67%] Linking C executable edge_hsm_x509_ut_exe
| [ 67%] Built target edge_hsm_x509_ut_exe
| [ 68%] Linking C executable edge_openssl_pki_ut_exe
| [ 68%] Built target edge_openssl_pki_ut_exe
| [ 69%] Linking C executable hsm_client_tpm_ut_exe
| [ 69%] Built target hsm_client_tpm_ut_exe
| Makefile:162: recipe for target 'all' failed
|
| --- stderr
| fatal: Not a git repository (or any parent up to mount point /home/jenkins/build)
| Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
| CMake Warning:
|   Manually-specified variables were not used by the project:
|
|     use_emulator
|
|
| /home/jenkins/build/build-x11-pico-imx7/tmp/work/cortexa7hf-neon-poky-linux-gnueabi/iotedge-cli/1.0.8-r0/recipe-sysroot-native/usr/lib/libssl.so: file not recognized: File format not recognized
| collect2: error: ld returned 1 exit status
| make[2]: *** [libiothsm.so.1.0.8] Error 1
| make[1]: *** [CMakeFiles/iothsm.dir/all] Error 2
| make[1]: *** Waiting for unfinished jobs....
| make: *** [all] Error 2
| thread 'main' panicked at '
| command did not execute successfully, got: exit code: 2
|
| build script failed, must exit now', /home/jenkins/build/build-x11-pico-imx7/tmp/work/cortexa7hf-neon-poky-linux-gnueabi/iotedge-cli/1.0.8-r0/cargo_home/bitbake/cmake-0.1.30/src/lib.rs:643:5
| stack backtrace:
|    0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
|              at src/libstd/sys/unix/backtrace/tracing/gcc_s.rs:39
|    1: std::sys_common::backtrace::print
|              at src/libstd/sys_common/backtrace.rs:70
|              at src/libstd/sys_common/backtrace.rs:58
|    2: std::panicking::default_hook::{{closure}}
|              at src/libstd/panicking.rs:200
|    3: std::panicking::default_hook
|              at src/libstd/panicking.rs:215
|    4: std::panicking::rust_panic_with_hook
|              at src/libstd/panicking.rs:478
|    5: std::panicking::continue_panic_fmt
|              at src/libstd/panicking.rs:385
|    6: std::panicking::begin_panic_fmt
|              at src/libstd/panicking.rs:340
|    7: cmake::fail
|    8: cmake::run
|    9: cmake::Config::build
|   10: build_script_build::main
|   11: std::rt::lang_start::{{closure}}
|   12: std::panicking::try::do_call
|              at src/libstd/rt.rs:49
|              at src/libstd/panicking.rs:297
|   13: __rust_maybe_catch_panic
|              at src/libpanic_unwind/lib.rs:87
|   14: std::rt::lang_start_internal
|              at src/libstd/panicking.rs:276
|              at src/libstd/panic.rs:388
|              at src/libstd/rt.rs:48
|   15: main
|   16: __libc_start_main
|   17: _start
|
| WARNING: exit code 101 from a shell command.
| ERROR: Function failed: do_compile (log file is located at /home/jenkins/build/build-x11-pico-imx7/tmp/work/cortexa7hf-neon-poky-linux-gnueabi/iotedge-cli/1.0.8-r0/temp/log.do_compile.78571)
ERROR: Task (/home/jenkins/build/sources/meta-iotedge/recipes-core/iotedge-cli/iotedge-cli_1.0.8.bb:do_compile) failed with exit code '1'
NOTE: Tasks Summary: Attempted 916 tasks of which 832 didn't need to be rerun and 1 failed.

Summary: 1 task failed:
  /home/jenkins/build/sources/meta-iotedge/recipes-core/iotedge-cli/iotedge-cli_1.0.8.bb:do_compile
Summary: There was 1 ERROR message shown, returning a non-zero exit code.
```

**Build System (please complete the following information):**
 - OS: Ubuntu 16.04 docker container provided by Technexion
 - Host: Ubuntu 18.04

**Target System (please complete the following information):**
 - Machine: NXP IMX7D (ARMv7 32bit)
 - OS: Yocto
 - Version 2.4 Rocko (linux 4.9)

**Additional context**
Image is working including Docker without the azure iot-edge components

**Comment by geoffrey-vl** [comment]
fixed by #35

**Comment by wlenzeder** [comment]
I can confirm that build for Yocto Sumo is fixed by #35

**Comment by micahl** [comment]
Closing
